src/file_watcher.rs

```diff
@@ -0,0 +1,1426 @@
+use crate::{
+    daemonise, enforce_singleton, into_recoverable, telemetry_config, EnvSetting, Result,
+    TelemetryError, WatcherResult,
+};
+
+use base64::{engine::general_purpose::STANDARD, DecodeError, Engine};
+use chrono::NaiveDateTime;
+use influxdb_line_protocol::LineProtocolBuilder;
+use nix::{
+    fcntl::{Flock, FlockArg},
+    sys::signal::{kill, Signal::SIGKILL},
+    unistd::{getpid, Pid},
+};
+use regex::{Captures, Regex};
+use reqwest::blocking::{Client, Request, RequestBuilder};
+use std::fs::ReadDir;
+use std::{
+    clone::Clone,
```

Comment:
hmm do we really need to import this here?

Reply:
Gah! Good catch. 4539e12f8768